plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -129,30 +130,39 @@ fullyApplyAndBetaReduce ::
   AppContext tyname name uni fun ann ->
   InlineM tyname name uni fun ann (Maybe (Term tyname name uni fun ann))
 fullyApplyAndBetaReduce info args0 = do
-  rhsBody <- liftDupable (let Done rhsBody = varRhsBody info in rhsBody)
-  let go ::
+      -- cannot use `liftDupable` here because then the variables will get renamed and we won't be
```

Comment:
The only reason the inliner worked before even though `rhsBody` was renamed is because the body only has free variables! See [Rename](https://github.com/input-output-hk/plutus/blob/ba1eb6b7dc3bf1cf3fa2adcb261bccb90adf7f06/plutus-core/plutus-core/src/PlutusCore/Rename.hs#L35).